lib/puppet/util/tagging.rb

```diff
@@ -1,7 +1,7 @@
 require 'puppet/util/tag_set'
 
 module Puppet::Util::Tagging
-  ValidTagRegex = /\A[0-9A-Za-z_][0-9A-Za-z_:.-]*\Z/
+  ValidTagRegex = /\A[[:word:]][[[:word:]]:.-]*\Z/
```

Comment:
Ah ah, you were referring to the `[[[:word:]]:.-]` being rewritable as `[[:word:]:.-]` - I see. 
 Yes, that's fine.  

I thought you were referring to the first `[[:word:]]` which cannot have the double brackets removed: - i.e. 

```
[1] pry(main)> "abc" =~ /[:word:]/
=> nil
[2] pry(main)> "abc" =~ /[[:word:]]/
=> 0
```

As for the modules ticket, I've been told that concat makes extensive use of tags, and without tag names fully supporting UTF-8, concat explodes - so this is the root issue (at least with how concat is currently implemented).  I haven't looked at the concat code nor am I aware if it can be reasonably rewritten without tags.